StartBonusActivation/handler.ts

```diff
@@ -197,22 +207,31 @@ const createBonusActivation = (
   bonusActivationModel: BonusActivationModel,
   fiscalCode: FiscalCode,
   dsu: Dsu
-): TaskEither<IResponseErrorInternal, BonusActivation> =>
-  fromQueryEither(() => {
-    // TODO: generate bonus code with provided algorithm
-    const bonusCode = "any code" as BonusCode & NonEmptyString;
-    const bonusActivation: NewBonusActivation = {
-      applicantFiscalCode: fiscalCode,
-      createdAt: new Date(),
-      dsuRequest: dsu,
-      id: bonusCode,
-      kind: "INewBonusActivation",
-      status: BonusActivationStatusEnum.PROCESSING
-    };
-    return bonusActivationModel.create(bonusActivation, fiscalCode);
-  }).mapLeft(err =>
+): TaskEither<IResponseErrorInternal, BonusActivation> => {
+  const shouldRepeat = (l: QueryError | Error) => {
+    // TODO: fix this condition
+    return !(l instanceof Error) && !!l.code && l.code === 409;
+  };
+
+  const lazyQueryTask = () =>
```

Comment:
naming: try not to name methods using implementation details (`tryCreateNewBonus()` would be better)